Modules/HTTP/Sources/HTTP/Middleware/ContentNegotiationMiddleware/ContentNegotiatonMiddleware.swift

```diff
@@ -3,154 +3,118 @@ public enum ContentNegotiationMiddlewareError : Error {
     case noSuitableSerializer
 }
 
-public struct ContentNegotiationMiddleware : Middleware {
-    public let types: [MediaTypeRepresentor.Type]
-    public let mode: Mode
+func parserTypes(for mediaType: MediaType, `in` types: [MediaTypeConvertible.Type]) -> [(MediaType, MapParser.Type)] {
```

Comment:
Is this function necessary? Might be best to inline it into a one-liner at the call site (L65)